src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -151,26 +152,36 @@ public IWebHost Build()
             }
             _webHostBuilt = true;
 
-            // Warn about deprecated environment variables
-            if (Environment.GetEnvironmentVariable("Hosting:Environment") != null)
+            var hostingServices = BuildCommonServices(out var hostingStartupErrors);
+            var applicationServices = hostingServices.Clone();
+            var hostingServiceProvider = hostingServices.BuildServiceProvider();
+            
+            if (_options.WriteStatusMessages)
             {
-                Console.WriteLine("The environment variable 'Hosting:Environment' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
-            }
+                // Warn about deprecated environment variables
+                if (Environment.GetEnvironmentVariable("Hosting:Environment") != null)
+                {
+                    Console.WriteLine("The environment variable 'Hosting:Environment' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
```

Comment:
Do we intend to ever remove these fallbacks? Also, should we log this instead of writing to the console?